contracts/bridge/SynapseBridge.sol

```diff
@@ -244,19 +279,71 @@ contract SynapseBridge is
             hasRole(NODEGROUP_ROLE, msg.sender),
             "Caller is not a node group"
         );
+
+        bool rateLimited = isRateLimited(address(token), amount);
+        if (rateLimited) {
+            rateLimiter.addToRetryQueue(
+                kappa,
+                abi.encodeWithSelector(
+                    this.retryWithdraw.selector,
+                    to,
+                    address(token),
+                    amount,
+                    fee,
+                    kappa
+                )
+            );
+            return;
```

Comment:
Do we want to adopt if/else pattern here of `if -> return` + `doWithdraw()`?

Reply:
Changed in [55487d3](https://github.com/synapsecns/synapse-contracts/pull/116/commits/55487d34fb5c4ee3d96a660f638b55e7c724b101)